dotCMS/src/main/java/com/dotcms/storage/FileStorageAPIImpl.java

```diff
@@ -0,0 +1,308 @@
+        this.objectReaderDelegate = objectReaderDelegate;
+    }
+
+    @Override
+    public void setObjectWriterDelegate(final ObjectWriterDelegate objectWriterDelegate) {
+        this.objectWriterDelegate = objectWriterDelegate;
+    }
+
+    @Override
+    public void setMetadataGenerator(final MetadataGenerator metadataGenerator) {
+        this.metadataGenerator = metadataGenerator;
+    }
+
+    public StorageProvider getStorageProvider() {
+        return this.storageProvider;
+    }
+
+    @Override
+    public Map<String, Object> generateRawBasicMetaData(final File binary) {
+
+        return this.generateBasicMetaData(binary, s -> true); // raw = no filter
+    }
+
+    @Override
+    public Map<String, Object> generateRawFullMetaData(final  File binary, long maxLength) {
+
+        return this.generateFullMetaData(binary, s -> true, maxLength); // raw = no filter
+    }
+
+    @Override
+    public Map<String, Object> generateBasicMetaData(final File binary, final Predicate<String> metaDataKeyFilter) {
+
+        final ImmutableSortedMap.Builder<String, Object> mapBuilder =
+                new ImmutableSortedMap.Builder<>(Comparator.naturalOrder());
+
+        if (this.validBinary(binary)) {
+
+            if (metaDataKeyFilter.test(TITLE_META_KEY)) {
+                mapBuilder.put(TITLE_META_KEY, binary.getName());
+            }
+
+            if (metaDataKeyFilter.test(PATH_META_KEY)) {
+                mapBuilder.put(PATH_META_KEY, binary.getAbsolutePath());
+            }
+
+            if (metaDataKeyFilter.test(LENGTH_META_KEY)) {
+                mapBuilder.put(LENGTH_META_KEY, binary.length());
+            }
+
+            if (metaDataKeyFilter.test(CONTENT_TYPE_META_KEY)) {
+                mapBuilder.put(CONTENT_TYPE_META_KEY, MimeTypeUtils.getMimeType(binary));
+            }
+
+            mapBuilder.put(MOD_DATE_META_KEY, System.currentTimeMillis());
+            mapBuilder.put(SHA226_META_KEY,   Try.of(()->FileUtil.sha256toUnixHash(binary)).getOrElse("unknown"));
+        }
+
+        return mapBuilder.build();
+    }
+
+    @Override
+    public Map<String, Object> generateFullMetaData(final File binary, final Predicate<String> metaDataKeyFilter,
+                                                    final long maxLength) {
+
+        final TreeMap<String, Object> metadataMap = new TreeMap<>(Comparator.naturalOrder());
+
+        try {
+
+            metadataMap.putAll(this.generateBasicMetaData(binary, metaDataKeyFilter));
+            final Map<String, Object> fullMetaDataMap = this.metadataGenerator.generate(binary, maxLength);
+            if (UtilMethods.isSet(fullMetaDataMap)) {
+                for (final Map.Entry<String, Object> entry : fullMetaDataMap.entrySet()) {
+
+                    if (metaDataKeyFilter.test(entry.getKey())) {
+
+                        metadataMap.put(entry.getKey(), entry.getValue());
+                    }
+                }
+            }
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)